data_model/src/isi.rs

```diff
@@ -924,13 +924,30 @@ mod transparent {
 
     isi! {
         /// Instruction to execute specified trigger
-        #[derive(Constructor, Display)]
+        #[derive(Display)]
         #[display(fmt = "EXECUTE `{trigger}`")]
-        #[serde(transparent)]
-        #[repr(transparent)]
         pub struct ExecuteTrigger {
             /// Id of a trigger to execute
             pub trigger: TriggerId,
+            /// Arguments to trigger execution
+            pub args: Option<JsonString>,
```

Comment:
```suggestion
            pub args: Vec<JsonString>,
```

why isn't it a list of args?

Reply:
Because it easily could be encoded as json array if required or as object to have named fields.